### PR TITLE
feat: add idp templates

### PR DESCRIPTION
**Pull request by eliobischof**

### CODE REVIEW

File: zitadel/v2/login_policy/funcs.go [hunk 1]
```diff
@@ -251,20 +249,42 @@ func create(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Dia
...
 }
 
+func getIDPOwnerType(ctx context.Context, client *mgmtclient.Client, id string) (idp.IDPOwnerType, error) {
+	ownertype := idp.IDPOwnerType_IDP_OWNER_TYPE_UNSPECIFIED
+	templateProvider, err := client.GetProviderByID(ctx, &management.GetProviderByIDRequest{Id: id})
+	err = helper.IgnoreIfNotFoundError(err)
+	if err != nil {
+		return idp.IDPOwnerType_IDP_OWNER_TYPE_UNSPECIFIED, err
+	}
+	ownertype = templateProvider.GetIdp().GetOwner()
```
Comment (adlerhurst) [inline zitadel/v2/login_policy/funcs.go:270]:
```suggestion
	templateProvider, err := client.GetProviderByID(ctx, &management.GetProviderByIDRequest{Id: id})
	err = helper.IgnoreIfNotFoundError(err)
	if err != nil {
		return idp.IDPOwnerType_IDP_OWNER_TYPE_UNSPECIFIED, err
	}
	ownertype := templateProvider.GetIdp().GetOwner()
```

File: zitadel/v2/org_idp_github/model.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+		ClientId:        m.ClientID.ValueString(),
+		ClientSecret:    m.ClientSecret.ValueString(),
+		Scopes:          m.Scopes,
+		ProviderOptions: m.toPbIdpOptions(),
+	}
+}
+
+func (m *tfModelSensitive) toPbUpdateGithubProviderRequest() *management.UpdateGitHubProviderRequest {
+	return &management.UpdateGitHubProviderRequest{
+		Id:              m.ID,
```
Comment (adlerhurst) [inline zitadel/v2/org_idp_github/model.go:52]: shouldn't it also be 
```suggestion
		Id:              m.ID.ValueString(),
```

**Review by adlerhurst — COMMENTED** [review]
looks good 👍

**Comment by eliobischof** [comment]
Closing in favor of #81

**Pull request closed without merging**